### DISCUSSION TITLE
gcsa vs. oauth2client - Client secrets must be for a web or installed app.

### DISCUSSION THREAD

**Discussion by parrela** [kuzmoyev/google-calendar-simple-api#125] [Q&A] [answered]
I'm a beginner so I hope this question isn't annoying:
I can get events from my google calendar using the original oauth2client tools, but when I try to use gcsa I get "Client secrets must be for a web or installed app." My credentials must be ok, so what am I doing wrong with such a simple program? Maybe the error indicates the credentials weren't found for some reason?

def get_easier_google_events(startdate, enddate):

    calendar = GoogleCalendar('xxxxxxx@gmail.com')
    for event in calendar:
        print(event)

**Comment by kuzmoyev** [comment, marked as answer]
Could you please post your oauth2client code that works for you?

**Reply by kuzmoyev** [reply]
If these are your credentials, you should delete this :) They should be private.
I was talking about the python code (without gcsa) that you've tried and were able to authenticate and retrieve the events

**Reply by kuzmoyev** [reply]
(I've deleted your comment. Please check if your credentials are anywhere in a public repository(ies) and make sure to delete them and add to .gitignore)

**Reply by parrela** [reply]
Thanks for helping this old fool... anyway here is the code that works. Mostly cribbed from stackoverflow:
```Python
def get_google_events(startdate, enddate):
    # Setup the Calendar API
    SCOPES = 'https://www.googleapis.com/auth/calendar.readonly'
    store = file.Storage('credentials.json')
    creds = store.get()
    if not creds or creds.invalid:
        flow = client.flow_from_clientsecrets('client_secret.json', SCOPES)
        creds = tools.run_flow(flow, store)
    service = build('calendar', 'v3', http=creds.authorize(Http()))

    # Call the Calendar API
    now = dt.datetime.utcnow().isoformat() + 'Z' # 'Z' indicates UTC time
    print('Getting the upcoming 10 events')
    events_result = service.events().list(calendarId='primary', timeMin=now,
                                          maxResults=100, singleEvents=True,
                                          orderBy='startTime').execute()
    events = events_result.get('items', [])

    if not events:
        print('No upcoming events found.')
    for event in events:
        start = event['start'].get('dateTime', event['start'].get('date'))
        print(start, event['summary'])
```

**Reply by kuzmoyev** [reply]
So the main difference is that `GoogleCalendar('xxxxxxx@gmail.com')` reads credentials from the `~/.credentials/credentials.json` by default. While this code gets them from the current directory (I suppose from the root of your project). You need to define the path to the file manualy: `GoogleCalendar('xxxxxxx@gmail.com', credentials_path='client_secret.json')`.
Please, let me know if this works or if you get any other error.

**Reply by parrela** [reply]
That does work. I could have found this in your docs, I guess. I was thrown by the error message that seemed to indicate I had credentials for the wrong type of application. Thanks.

**Reply by kuzmoyev** [reply]
Great. Glad it helped.
That means you have a `~/.credentials/credentials.json` file in your home directory that gcsa tried to use by default and failed.
